src/gensxg.cc

```diff
@@ -0,0 +1,307 @@
+#include <time.h>
+#include <unistd.h>
+
+#include <openssl/pem.h>
+
+#include <memory>
+#include <string>
+#include <iostream>
+
+#include "libsxg.h"
+
+struct Options {
+ private:
+  static EVP_PKEY* LoadPrivateKey(const std::string& filepath) {
+    FILE* const keyfile = fopen(filepath.c_str(), "r");
+    if (keyfile == nullptr) {
+      return nullptr;
+    }
+    EVP_PKEY* private_key =
+        PEM_read_PrivateKey(keyfile, nullptr, nullptr, nullptr);
+    fclose(keyfile);
+    return private_key;
+  }
+
+  static X509* LoadX509Cert(const std::string& filepath) {
+    FILE* certfile = fopen(filepath.c_str(), "r");
+    if (certfile == nullptr) {
+      return nullptr;
+    }
+    char passwd[] = "";
+    X509* cert = PEM_read_X509(certfile, 0, 0, passwd);
+    fclose(certfile);
+    return cert;
+  }
+
+  static EVP_PKEY* LoadEd25519Pubkey(const std::string& filepath) {
+    FILE* keyfile = fopen(filepath.c_str(), "r");
+    if (!keyfile) {
+      return nullptr;
+    }
+    EVP_PKEY* public_key = PEM_read_PUBKEY(keyfile, nullptr, nullptr, nullptr);
+    fclose(keyfile);
+    return public_key;
+  }
+
+ public:
+  Options(int argc, const char** argv)
+      : help(false),
+        content_type("text/html"),
+        date(0),
+        expire(0),
+        mi_record_size(4096),
+        output("out.sxg") {
```

Comment:
I think the standard output is more natural as a fallback. Also there seems no way to change the output file ― it's clearly inconvenient for users.